tests/test_symbolic_coefficients.py

```diff
@@ -105,6 +105,21 @@ def test_function_coefficients_xderiv(self):
         op = Operator(eqn)
         op()
 
+    @pytest.mark.parametrize('order', [2, 4, 6, 8])
+    def test_function_coefficients_xderiv(self, order):
+        p = Dimension('p')
+
+        grid = Grid(shape=(51, 51, 51))
+        x, y, z = grid.dimensions
+
+        f = Function(name='f', grid=grid, space_order=order)
+        w = Function(name='w', space_order=0, shape=(*grid.shape, order + 1),
+                     dimensions=(x, y, z, p))
+
+        expr0 = f.dx(w=w).dy(w=w).evaluate
```

Comment:
Minor suggestion: Can you supply different weights for each derivative?

Reply:
No that's the point of the test. If you provide different ones it already works but if you provide the same one it was not finding the stencil dimension in `w` properly for the second one

Reply:
Ah ok